Github Actions. Could not determine the Composer version

## Bug report

| Question    | Answer
| ------------| ---------------
| Box version | Phar. 4.3.8@5534406 2023-03-17 08:56:09 UTC
| PHP version | 8.2.8
| Platform with version | GitHub Actions. Ubuntu 22.04.2 LTS
| Github Repo | https://github.com/JBZoo/Composer-Diff


Hello, 
I really like your tool and use it for many of my projects.

Recently I ran into a problem that I could not solve on my own. I spent quite a lot of time on it and decided to ask for help from you.

I really want to upgrade from box v3.16 to v4.3

On compilation, I see the error only(!) in GitHub Actions environment. On my local machine (MacOs Venture) it 100% works fine.

I prepared a test pipeline https://github.com/JBZoo/Composer-Diff/blob/18-autoloader/.github/workflows/box-project-issue.yml#L47-L66

You can see the result here
https://github.com/JBZoo/Composer-Diff/actions/runs/5500221650/jobs/10022985785

My guess the regexp works wrong here
https://github.com/box-project/box/blob/4.3.8/src/Composer/ComposerOrchestrator.php#L99-L103

I think it should take into account the "space" char before the version date.

Unfortunately, I have not found an option to bypass this check.

On the old version (3.16) of the box there was no such test of composer version. 



-----------


My exact software versions are here - https://github.com/JBZoo/Composer-Diff/actions/runs/5500221650/jobs/10022985785#step:5:1

```
1. which php
/usr/bin/php

2. php -v
PHP 8.2.8 (cli) (built: Jul  8 2023 07:10:21) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.8, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.8, Copyright (c), by Zend Technologies

3. which composer
/usr/local/bin/composer

4. composer --version
Composer version 2.5.8 2023-06-09 17:13:21

5. /usr/local/bin/composer --version
Composer version 2.5.8 2023-06-09 17:13:21

6. php ./vendor/bin/box.phar --version
Box version 4.3.8@5534406 2023-03-17 08:56:09 UTC
```

<details>
 <summary>box.json.dist</summary>
 
 ```json
 {
    "banner"      : [
        "JBZoo Toolbox - Composer-Diff",
        "",
        "This file is part of the JBZoo Toolbox project.",
        "For the full copyright and license information, please view the LICENSE",
        "file that was distributed with this source code.",
        "",
        "@package    Composer-Diff",
        "@license    MIT",
        "@copyright  Copyright (C) JBZoo.com, All rights reserved.",
        "@link       https://github.com/JBZoo/Composer-Diff"
    ],

    "output"      : "build/composer-diff.phar",

    "directories" : ["src"],
    "files"       : ["composer-diff.php"],
    "git-version" : "git-version",

    "finder"      : [
        {"in" : "vendor"}
    ]
}

 ```
</details>

<details>
 <summary>Output</summary>
 
 ```bash
php ./vendor/bin/box.phar compile -v

    ____
   / __ )____  _  __
  / __  / __ \| |/_/
 / /_/ / /_/ />  <
/_____/\____/_/|_|


Box version 4.3.8@5534406 2023-03-[17](https://github.com/JBZoo/Composer-Diff/actions/runs/5500211945/jobs/10022969632#step:6:18) 08:56:09 UTC

 // Loading the configuration file "/home/runner/work/Composer-Diff/Composer-Diff/box.json.dist".                       

🔨  Building the PHAR "/home/runner/work/Composer-Diff/Composer-Diff/build/composer-diff.phar"

? Checking Composer compatibility
    > '/usr/local/bin/composer' '--version'

In ComposerOrchestrator.php line 52:

  [RuntimeException]                         
  Could not determine the Composer version.

Exception trace:
  at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/src/Composer/ComposerOrchestrator.php:52
 _HumbugBox89e0edca2b33\KevinGH\Box\Composer\ComposerOrchestrator::getVersion() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/src/Composer/ComposerOrchestrator.php:32
 _HumbugBox89e0edca2b33\KevinGH\Box\Composer\ComposerOrchestrator::checkVersion() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/src/Console/Command/Compile.php:188
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile::checkComposerVersion() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/src/Console/Command/Compile.php:140
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile->createPhar() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/src/Console/Command/Compile.php:126
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile->execute() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/vendor/fidry/console/src/Command/SymfonyCommand.php:61
 _HumbugBox89e0edca2b33\Fidry\Console\Command\SymfonyCommand->execute() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/vendor/symfony/console/Command/Command.php:168
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Command\Command->run() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/vendor/symfony/console/Application.php:681
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Application->doRunCommand() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/vendor/symfony/console/Application.php:226
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Application->doRun() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/vendor/symfony/console/Application.php:122
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Application->run() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/vendor/fidry/console/src/Application/ApplicationRunner.php:[34](https://github.com/JBZoo/Composer-Diff/actions/runs/5500221650/jobs/10022985785#step:6:35)
 _HumbugBox89e0edca2b33\Fidry\Console\Application\ApplicationRunner->run() at phar:///home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar/bin/box:47
 require() at /home/runner/work/Composer-Diff/Composer-Diff/vendor/bin/box.phar:18

compile [--debug] [--no-parallel] [--no-restart] [--dev] [--no-config] [--with-docker] [--composer-bin COMPOSER-BIN] [--allow-composer-check-failure] [-c|--config CONFIG] [-d|--working-dir WORKING-DIR]

Error: Process completed with exit code 1.
 ```
</details>

Thank you very much in advance.
I would be very happy to get any help or advice from you.

--- a/src/Composer/ComposerOrchestrator.php
+++ b/src/Composer/ComposerOrchestrator.php
@@ -75,7 +75,7 @@ final class ComposerOrchestrator
         $logger = new CompilerLogger($io ?? IO::createNull());
 
         $composerExecutable = $composerBin ?? self::retrieveComposerExecutable();
-        $getVersionProcess = new Process([$composerExecutable, '--version']);
+        $getVersionProcess = new Process([$composerExecutable, '--version', '--no-ansi']);
 
         $logger->log(
             CompilerLogger::CHEVRON_PREFIX,

--- a/tests/Console/Command/CompileTest.php
+++ b/tests/Console/Command/CompileTest.php
@@ -858,7 +858,7 @@ class CompileTest extends FileSystemTestCase
 
             ? Removing the existing PHAR "/path/to/tmp/test.phar"
             ? Checking Composer compatibility
-                > '/usr/local/bin/composer' '--version'
+                > '/usr/local/bin/composer' '--version' '--no-ansi'
                 > 2.5.0 (Box requires ^2.2.0)
                 > Supported version detected
             ? Registering compactors
@@ -986,7 +986,7 @@ class CompileTest extends FileSystemTestCase
 
             ? Removing the existing PHAR "/path/to/tmp/test.phar"
             ? Checking Composer compatibility
-                > '/usr/local/bin/composer' '--version'
+                > '/usr/local/bin/composer' '--version' '--no-ansi'
                 > 2.5.0 (Box requires ^2.2.0)
                 > Supported version detected
             ? Registering compactors

